### PR TITLE
core: fade out improvements

### PR DESCRIPTION
**Pull request by PaideiaDilemma**
Wanted the fade out animation for grace unlocks as well.
But also:
 - disables input once the fade has started
 - renders after the fade out started (needed for grace unlock, fine for normal unlocks as well I guess)
 - don't disable the fade out for other compositors (not really needed)

### CODE REVIEW

**Comment by vaxerski** [comment]
> don't disable the fade out for other compositors (not really needed)

Needed. Other compositors render a solid background behind it, and the fade out is idiotic in that case.

**Comment by PaideiaDilemma** [comment]
yeah I just though it is fine because fade in kinda behaves the same, it does not look too bad (checked on sway) and you can turn it off anyways.

But I will undo it :)

File: src/core/hyprlock.cpp [hunk 1]
```diff
@@ -530,6 +530,24 @@ void CHyprlock::run() {
     Debug::log(LOG, "Reached the end, exiting");
 }
 
+void CHyprlock::unlock() {
```
Comment (vaxerski) [inline src/core/hyprlock.cpp:533]: naming gets confusing here, `unlockChecked` and `unlockUnchecked`, maybe? Or something?

Comment (PaideiaDilemma) [reply src/core/hyprlock.cpp:533]: Agree. How about renaming `lockSession` and `unlockSession` to smth like `acquireSessionLock` and `releaseSessionLock`?

Comment (vaxerski) [reply src/core/hyprlock.cpp:533]: sure

**Review by vaxerski — APPROVED** [review]
lgtm

**Comment by bvr-yr** [comment]
i'm also working on some fixes and have a question here
do we need `success_color`? or just fading back to outer is enough
because currently stuck `check_color` looks confusing when unlocking
probably just make it as an option and default to -1 = outer?

**Comment by vaxerski** [comment]
idk

**Pull request merged**